src/main/java/org/tron/program/FullNode.java

```diff
@@ -104,4 +125,46 @@ public static void shutdown(final Application app) {
     logger.info("********register application shutdown hook********");
     Runtime.getRuntime().addShutdownHook(new Thread(app::shutdown));
   }
+
+  public static void cache(TronApplicationContext context) {
+    long start = System.currentTimeMillis();
+    long headNum = context.getBean(DynamicPropertiesStore.class).getLatestBlockHeaderNumber();
+    long recentBlockCount = Streams.stream(context.getBean(RecentBlockStore.class).getRevokingDB()).count();
+    Manager manager = context.getBean(Manager.class);
+    Map<Key, Long> trxIds = new ConcurrentHashMap<>();
+    ListeningExecutorService service = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(50));
+    List<ListenableFuture<?>> futures = new ArrayList<>();
+    AtomicLong atomicLong = new AtomicLong(0);
+    LongStream.rangeClosed(headNum - recentBlockCount + 1, headNum).forEach(
+        blockNum -> futures.add(service.submit(() -> {
+          try {
+            atomicLong.incrementAndGet();
+            BlockCapsule blockCapsule = manager.getBlockByNum(blockNum);
+            blockCapsule.getTransactions().stream()
+                .map(tc -> tc.getTransactionId().getBytes())
+                .map(bytes -> Maps.immutableEntry(Key.of(bytes), blockNum))
+                .forEach(e -> trxIds.put(e.getKey(), e.getValue()));
+          } catch (ItemNotFoundException e) {
+            e.printStackTrace();
+          } catch (BadItemException e) {
+            e.printStackTrace();
+          }
+        })));
+    ListenableFuture<?> future = Futures.allAsList(futures);
+    try {
+      future.get();
+    } catch (InterruptedException e) {
+      e.printStackTrace();

```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Use a logger to log this exception. ([squid:S1148](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS1148))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1891&issues=AWf0QKUgrt4Kct_4mYKy&open=AWf0QKUgrt4Kct_4mYKy)

[ISSUE_KEY]: AWf0QKUgrt4Kct_4mYKy